code/src/App.js

```diff
@@ -1,9 +1,78 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
+import { API_URL, API_LIKES_URL } from './utils/Api'
+import { Form } from './components/Form'
+import { ThoughtList } from './components/ThoughtList'
 
 export const App = () => {
+  const [thoughts, setThoughts] = useState([])
+  const [newThought, setNewThought] = useState('')
+  const [counter, setCounter] = useState(0)
+
+  // calling useEffect to fetch all messages after component get mounted
+  useEffect(() => {
+    fetchThoughts()
+    // return () => {
+    //   console.log('I will be unmounted')
+    // }
+  }, [])
+
+  const fetchThoughts = () => {
+    fetch(API_URL)
+      .then((res) => res.json())
+      .then((data) => setThoughts(data)) // saves data in the state
+  }
+
+  const onNewThoughtChange = (event) => {
+    setNewThought(event.target.value)
+    setCounter(event.target.value.length)
+  }
+
+  // stops refreshing submittion of the form and executes:
+  const handleFormSumbit = (event) => {
+    event.preventDefault()
+
+    const options = {
+      method: 'POST', // hämta eller skicka
+      headers: {
+        // for the API to know what type of data we will send. headers: helps the server understand what will be sent, what i will receive and what i will get back what language, if its pictures osv.
```

Comment:
Great comments on what the different parts do! Makes it really easy to understand what's going on!